src/django/api/management/commands/batch_process.py

```diff
@@ -0,0 +1,110 @@
+                                'One of "parse", "geocode", "match"')
+        group.add_argument('-l', '--list-id',
+                           required=True,
+                           help='The id of the facility list to process.')
+
+    def handle(self, *args, **options):
+        action = options['action']
+        list_id = options['list_id']
+
+        try:
+            # Crash if invalid action specified
+            if action not in ACTIONS.keys():
+                self.stderr.write('Validation Error: Invalid action "{}". '
+                                  'Must be one of "parse", "geocode", "match".'
+                                  .format(action))
+                sys.exit(1)
+
+            process = ACTIONS[action]
+            facility_list = FacilityList.objects.get(pk=list_id)
+
+            row_index = os.environ.get('AWS_BATCH_JOB_ARRAY_INDEX')
+            if row_index:
+                items = FacilityListItem.objects.filter(
+                    facility_list=facility_list,
+                    row_index=row_index)
+            else:
+                items = FacilityListItem.objects.filter(
+                    facility_list=facility_list)
+
+            result = {
+                'success': 0,
+                'failure': 0,
+            }
+
+            # Process all items, save affected items, facilities, matches,
+            # and tally successes and failures
+            for item in items:
+                try:
+                    with transaction.atomic():
+                        if action == ProcessingAction.MATCH:
+                            facility, match = process(item)
+                            item.save()
+                            facility.save()
+                            match.save()
+                        else:
+                            process(item)
+                            item.save()
+
+                        if item.status == FacilityListItem.ERROR:
+                            result['failure'] += 1
+                        else:
+                            result['success'] += 1
+                except ValueError as e:
+                    self.stderr.write('Value Error: {}'.format(e))
+                    result['failure'] += 1
+                except DatabaseError as e:
+                    self.stderr.write('Database Error: {}'.format(e))
+                    sys.exit(1)
+
+            # Print successes
+            if result['success'] > 0:
+                self.stdout.write(
+                    self.style.SUCCESS(
+                        '{}: {} successes'.format(
+                            action, result['success'])))
+
+            # Print failures
+            if result['failure'] > 0:
+                self.stdout.write(
+                    self.style.ERROR(
+                        '{}: {} failures'.format(
+                            action, result['failure'])))
+
+        except FacilityList.DoesNotExist:
+            self.stderr.write('Validation Error: '
```

Comment:
Should we also `sys.exit(1)` here?

Reply:
I think so. It makes sense to fail the Job the AWS Batch level if one of the arguments was wrong.

Reply:
Should I do the same for `CommandError` beneath this?